tokio-rustls/src/lib.rs

```diff
@@ -126,6 +146,122 @@ impl TlsAcceptor {
+    #[inline]
+    pub fn new(acceptor: rustls::Acceptor, io: IO) -> Self {
+        Self {
+            acceptor,
+            buf: BytesMut::from(&[0; 512][..]),
+            used: 0,
+            io: Some(io),
+        }
+    }
+}
+
+impl<IO> Future for LazyConfigAcceptor<IO>
+where
+    IO: AsyncRead + AsyncWrite + Unpin,
+{
+    type Output = Result<Accepted<IO>, io::Error>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let this = self.get_mut();
+        while this.acceptor.wants_read() {
+            let io = match this.io.as_mut() {
+                Some(io) => io,
+                None => {
+                    return Poll::Ready(Err(io::Error::new(
+                        io::ErrorKind::Other,
+                        "acceptor cannot be polled after acceptance",
+                    )))
+                }
+            };
+
+            let mut buf = ReadBuf::new(&mut this.buf);
+            buf.advance(this.used);
+            if let Err(err) = ready!(Pin::new(io).poll_read(cx, &mut buf)) {
+                return Poll::Ready(Err(err));
+            }
+
+            let read = match this.acceptor.read_tls(&mut buf.filled()) {
+                Ok(read) => read,
+                Err(err) => return Poll::Ready(Err(err)),
+            };
+
+            let filled = buf.filled().len();
+            if read < filled {
+                this.buf.copy_within(read.., 0);
+                this.used = filled - read;
+            }
+
+            match this.acceptor.accept() {
+                Ok(Some(accepted)) => {
+                    let io = this.io.take().unwrap();
+                    return Poll::Ready(Ok(Accepted { accepted, io }));
+                }
+                Ok(None) => continue,
+                Err(err) => {
+                    return Poll::Ready(Err(io::Error::new(io::ErrorKind::InvalidInput, err)))
+                }
+            }
+        }
+
+        Poll::Pending
+    }
+}
+
+pub struct Accepted<IO> {
+    accepted: rustls::Accepted,
+    io: IO,
+}
+
+impl<IO> Accepted<IO>
+where
+    IO: AsyncRead + AsyncWrite + Unpin,
+{
+    pub fn client_hello(&self) -> rustls::ClientHello<'_> {
+        self.accepted.client_hello()
+    }
+
+    pub fn into_stream<F>(self, config: Arc<ServerConfig>) -> Accept<IO> {
+        self.into_stream_with(config, |_| ())
+    }
+
+    pub fn into_stream_with<F>(self, config: Arc<ServerConfig>, f: F) -> Accept<IO>
+    where
+        F: FnOnce(&mut ServerConnection),
+    {
+        let mut conn = match self.accepted.into_connection(config) {
+            Ok(conn) => conn,
+            Err(error) => {
+                return Accept(MidHandshake::Error {
+                    io: self.io,
+                    // TODO(eliza): should this really return an `io::Error`?
+                    // Probably not...
+                    error: io::Error::new(io::ErrorKind::Other, error),
+                });
+            }
+        };
+        f(&mut conn);
```

Comment:
Whats the use-case for the function? All the user to set further properties on the connection?

Reply:
No clue, I just copied what `TlsAcceptor::accept_with()` does.